source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc

```diff
@@ -417,6 +418,20 @@ void DnsCacheImpl::finishResolve(const std::string& host,
     return primary_host_it->second.get();
   }();
 
+  std::string details_with_maybe_trace = std::string(details);
+  if (runtime_.snapshot().getBoolean("envoy.enable_dfp_dns_trace", false)) {
+    if (primary_host_info != nullptr && primary_host_info->active_query_ != nullptr) {
+      absl::ReaderMutexLock reader_lock{&primary_hosts_lock_};
+      std::string trace = primary_host_info->active_query_->getTraces().empty()
+                              ? ""
+                              : absl::StrJoin(primary_host_info->active_query_->getTraces(), ",");
+      if (!trace.empty()) {
+        details_with_maybe_trace = absl::StrCat(details, ":", trace);
+      }
+      primary_host_info->active_query_->removeTraces();
```

Comment:
I'm a little bit concerned about the getaddrinfo code being in a different thread (so we may end up with somewhat unexpected results) but I think given this is runtime off and we're likely the only users, it's probably fine.